app/slave/cluster_slave.py

```diff
@@ -56,6 +58,29 @@ def __init__(self, port, host, num_executors=10):
         self._build_teardown_coin = None
         self._base_executor_index = None
 
+        self._heartbeat_count = 0
+        self._heartbeat_count_threshold = Configuration['heartbeat_count_threshold']
+        self._heartbeat_frequency = Configuration['heartbeat_frequency']
+        self.scheduler = BackgroundScheduler()
+        self._heartbeat_job = None
+        self._logger.info('Heartbeat will run every {} seconds'.format(self._heartbeat_frequency))
+
+    def configure_heartbeat(self):
+        self._heartbeat_job = self.scheduler.add_job(self.heartbeat, 'interval', seconds=self._heartbeat_frequency)
+        return self.scheduler
```

Comment:
Same comment about not needing to return the scheduler, and about inlining this into the constructor